### PR TITLE
Fix clang-CUDA warning in 2D memory allocation

### LINKED ISSUE #1000: Clang as CUDA Device Compiler throws a Compiler error
Error:
```bash
In file included from /alpaka/include/alpaka/alpaka.hpp:136:
/alpaka/include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404:50: error: implicit conversion changes signedness: 'const long' to 'std::size_t' (aka 'unsigned long') [-Werror,-Wsign-conversion]
                        std::size_t pitchBytes = widthBytes;
                                    ~~~~~~~~~~   ^~~~~~~~~~
/alpaka/include/alpaka/mem/buf/Traits.hpp:128:23: note: in instantiation of function template specialization 'alpaka::mem::buf::traits::Alloc<unsigned int, std::integral_constant<unsigned long, 2>, long, alpaka::dev::DevUniformCudaHipRt, void>::alloc<alpaka::vec::Vec<std::integral_constant<unsigned long, 2>, long> >' requested here
                    ::alloc(
                      ^
/alpaka/test/unit/mem/p2p/src/P2P.cpp:77:5: note: in instantiation of function template specialization 'testP2P<alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, long> >' requested here
    testP2P<Acc>( extent );
    ^
/alpaka/test/unit/mem/p2p/src/P2P.cpp:62:1: note: in instantiation of function template specialization '____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____F_U_N_C____0<alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, long> >' requested here
TEMPLATE_LIST_TEST_CASE( "memP2PTest", "[memP2P]", alpaka::test::acc::TestAccs)
^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:17353:40: note: expanded from macro 'TEMPLATE_LIST_TEST_CASE'
#define TEMPLATE_LIST_TEST_CASE( ... ) INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE(__VA_ARGS__)
                                       ^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:1186:130: note: expanded from macro 'INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE'
        INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE_2( INTERNAL_CATCH_UNIQUE_NAME( ____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____ ), INTERNAL_CATCH_UNIQUE_NAME( ____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____F_U_N_C____ ), Name, Tags, TmplList )
                                                                                                                                 ^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:448:46: note: expanded from macro 'INTERNAL_CATCH_UNIQUE_NAME'
#  define INTERNAL_CATCH_UNIQUE_NAME( name ) INTERNAL_CATCH_UNIQUE_NAME_LINE( name, __COUNTER__ )
                                             ^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:446:55: note: expanded from macro 'INTERNAL_CATCH_UNIQUE_NAME_LINE'
#define INTERNAL_CATCH_UNIQUE_NAME_LINE( name, line ) INTERNAL_CATCH_UNIQUE_NAME_LINE2( name, line )
                                                      ^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:445:56: note: expanded from macro 'INTERNAL_CATCH_UNIQUE_NAME_LINE2'
#define INTERNAL_CATCH_UNIQUE_NAME_LINE2( name, line ) name##line
                                                       ^
<scratch space>:238:1: note: expanded from here
____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____F_U_N_C____0
^
/alpaka/test/unit/mem/p2p/src/P2P.cpp:62:1: note: in instantiation of member function '(anonymous namespace)::ns_____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____1::____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____1<alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1>, long>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 1>, long>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 1>, long>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 2>, long>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, long>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 2>, long>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 3>, long>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 3>, long>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 3>, long>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned long>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 1>, unsigned long>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 1>, unsigned long>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 2>, unsigned long>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, unsigned long>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 2>, unsigned long>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 3>, unsigned long>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 3>, unsigned long>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 3>, unsigned long>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1>, int>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 1>, int>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 1>, int>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 2>, int>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, int>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 2>, int>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 3>, int>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 3>, int>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 3>, int>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned int>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 1>, unsigned int>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 1>, unsigned int>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 2>, unsigned int>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, unsigned int>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 2>, unsigned int>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 3>, unsigned int>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 3>, unsigned int>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 3>, unsigned int>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1>, short>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 1>, short>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 1>, short>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 2>, short>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, short>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 2>, short>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 3>, short>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 3>, short>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 3>, short>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned short>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 1>, unsigned short>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 1>, unsigned short>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 2>, unsigned short>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 2>, unsigned short>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 2>, unsigned short>, alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 3>, unsigned short>, alpaka::acc::AccGpuUniformCudaHipRt<std::integral_constant<unsigned long, 3>, unsigned short>, alpaka::acc::AccGpuCudaRt<std::integral_constant<unsigned long, 3>, unsigned short> >::reg_tests' requested here
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:17353:40: note: expanded from macro 'TEMPLATE_LIST_TEST_CASE'
#define TEMPLATE_LIST_TEST_CASE( ... ) INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE(__VA_ARGS__)
                                       ^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:1186:9: note: expanded from macro 'INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE'
        INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE_2( INTERNAL_CATCH_UNIQUE_NAME( ____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____ ), INTERNAL_CATCH_UNIQUE_NAME( ____C_A_T_C_H____T_E_M_P_L_A_T_E____T_E_S_T____F_U_N_C____ ), Name, Tags, TmplList )
        ^
/alpaka/test/catch_main/../../thirdParty/catch2/include/catch2/catch.hpp:1177:19: note: expanded from macro 'INTERNAL_CATCH_TEMPLATE_LIST_TEST_CASE_2'
                t.reg_tests();                                        \
                  ^
```

CMake config

```bash
cmake -DBOOST_ROOT=/opt/boost/1.67.0/ -DBOOST_LIBRARYDIR="/opt/boost/1.67.0/lib" -DCMAKE_BUILD_TYPE=Release -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE=ON -DALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE=OFF -DALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE=OFF -DALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE=OFF -DALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE=OFF -DALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE=OFF -DALPAKA_ACC_CPU_BT_OMP4_ENABLE=OFF -DALPAKA_ACC_GPU_CUDA_ENABLE=ON -DALPAKA_CUDA_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++-9 ..
```

Testet environments:
 - Ubuntu 18.04 + CUDA 10.1 + Boost 1.67.0 + Clang 9
 - Ubuntu 18.04 + CUDA 10.1 + Boost 1.73.0 + Clang 10

### PR DESCRIPTION
**Pull request by BenjaminW3**
Fixes #1000

### CODE REVIEW

File: test/common/include/alpaka/test/dim/TestDims.hpp [hunk 1]
```diff
@@ -34,9 +24,7 @@ namespace alpaka
             using TestDims =
                 std::tuple<
                     alpaka::dim::DimInt<1u>
-#if !defined(ALPAKA_CUDA_CI)
                     ,alpaka::dim::DimInt<2u>
-#endif
```
Comment (BenjaminW3) [inline test/common/include/alpaka/test/dim/TestDims.hpp:39]: We did not test the 2D CUDA case before.
The reason was that the builds timed out years ago. This seems to be no issue anymore.

File: include/alpaka/mem/buf/BufUniformCudaHipRt.hpp [hunk 2]
```diff
@@ -401,7 +401,7 @@ namespace alpaka
 
 
                         void * memPtr = nullptr;
-                        std::size_t pitchBytes = widthBytes;
+                        std::size_t pitchBytes = 0u;
```
Comment (BenjaminW3) [inline include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404]: `pitchBytes` is overwritten with the real value later so there is no need to initialize it with a best guess.

**Comment by BenjaminW3** [comment]
@psychocoderHPC Can we merge this irrespective of the failing HIP builds and the known flaky mac build?

**Review by sbastrakov — DISMISSED** [review]

File: include/alpaka/mem/buf/BufUniformCudaHipRt.hpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404]: Please revert this change. I think I fixed this in the past in the case the followed if body is not accessed the pitch would be zero. This is wrong.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: include/alpaka/mem/buf/BufUniformCudaHipRt.hpp (hunk 2 above)
Comment (sbastrakov) [reply include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404]: Ah, now i also remember something like that.

Comment (psychocoderHPC) [reply include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404]: Ohh wait this is fine because the memory pointer points to `nullptr`

Comment (BenjaminW3) [reply include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404]: Why does the pitch have to be !=0 when the size of the allocation is 0?
For the 3D case we set the pitch to zero as well in the exact same case.

Comment (psychocoderHPC) [reply include/alpaka/mem/buf/BufUniformCudaHipRt.hpp:404]: It was a wrong thinking on my side. All is fine with this change, if size is zero the pitch can be zero too.

**Comment by psychocoderHPC** [comment]
> @psychocoderHPC Can we merge this irrespective of the failing HIP builds and the known flaky mac build?

I am currently working on #1006, please let us wait.

**Comment by BenjaminW3** [comment]
I rebased onto the fixed develop. Please approve!

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**